lib/src/db/model_db_impl.dart

```diff
@@ -513,7 +514,8 @@ class _ExpandoModelDescription extends _ModelDescription<ExpandoModel> {
         model.additionalProperties[key] = value;
       }
     });
-    return model;
+    // TODO: check if there is a more elegant solution than this
+    return model as T;
```

Comment:
Would it address this TODO if we changed
```dart
T decodeEntity<T extends Model>(ModelDBImpl db, Key key, ds.Entity entity) {
  ExpandoModel model = super.decodeEntity(db, key, entity);
  ...
  // TODO: check if there is a more elegant solution than this
  return model as T;
}
```
to
```dart
T decodeEntity<T extends ExpandoModel>(ModelDBImpl db, Key key, ds.Entity entity) {
  T model = super.decodeEntity<T>(db, key, entity);
  ...
  return mode;
}
```
?

Reply:
The problem is, that using `T` instead of `ExpandoModel` means that there is no `additionalProperties` attribute, that is used just below.

Reply:
When using `T` and having a type-bound of `T extends ExpandoModel` then we should have all the properties of `ExpandoModel`.

Reply:
Oh sorry. I didn't see that you meant extending `ExpandoModel`. 

Well that would work in that specific case, but in every place this function is used we then get a type error because `T extends Model` is not `T extends ExpandoModel`

Reply:
Aha. Okay, so that doesn't work. Makes sense.